src/main/kotlin/org/rust/lang/refactoring/extractFunction/RsExtractFunctionConfig.kt

```diff
@@ -65,25 +66,65 @@ class RsExtractFunctionConfig private constructor(
     var visibilityLevelPublic: Boolean = false,
     val parameters: List<Parameter>
 ) {
+
+    private val typeParameters: List<RsTypeParameter>
+        get() {
+            val bounds = containingFunction.typeParameters.associate {
+                it.declaredType to it.bounds.flatMap {
+                    it.bound.traitRef?.path?.typeArgumentList?.typeReferenceList ?: emptyList()
+                }.mapNotNull { it.type }
+            }
+            return containingFunction.typeParameters.filter {
+                val typeParameter = it.declaredType
+                (parameters.mapNotNull { it.type } + listOf(returnValue).mapNotNull { it?.type }).any {
+                    it.types().contains(typeParameter) || it.dependTypes(bounds).contains(typeParameter)
+                }
+            }
+        }
+
+    private val typeParametersText: String
+        get() = typeParameters
+            .takeIf { it.isNotEmpty() }
+            ?.joinToString(separator = ",", prefix = "<", postfix = ">") { it.text }
+            ?: ""
+
+    private val whereClausesText: String
+        get() {
+            val whereTypes = containingFunction.whereClause?.wherePredList?.mapNotNull {
+                it.typeReference?.type
+            } ?: return ""
+            return typeParameters
+                .mapNotNull {
+                    if (!whereTypes.contains(it.declaredType) || it.bounds.isEmpty())
+                        null
+                    else
+                        "${it.text}:${it.bounds.joinToString("+") { it.text }}"
```

Comment:
Hm, it doesn't support the following case
```rust
fn foo<T: Debug>(t: T) where T: Default {
    <selection>println!("{}", t);</selection>
}
```
Probably it is because `it.bounds` returns all bounds not only from where clause